src/components/NodeList/index.tsx

```diff
@@ -4,71 +4,82 @@ import type {
   QueryParamConfigMap,
   SetQuery,
 } from "use-query-params";
-import type {
-  GridFilterModel,
-  GridRowClassNameParams,
-  GridValueFormatterParams,
-  GridColDef,
-  GridRenderCellParams,
-} from "@mui/x-data-grid";
+import {
+  useMaterialReactTable,
+  MaterialReactTable,
+  type MRT_ColumnDef,
+} from 'material-react-table';
 import Link from '@mui/material/Link';
 
 import type { Node } from "../../lib/paddles.d";
 import { formatDate } from "../../lib/utils";
-import DataGrid from "../DataGrid";
+import useDefaultTableOptions from "../../lib/table";
 
 
-export const columns: GridColDef[] = [
+export const columns: MRT_ColumnDef<Node>[] = [
   {
-    field: "name",
-    width: 100,
-    renderCell: (params: GridRenderCellParams) => {
-      return <Link href={`/nodes/${params.value}/`} color="inherit">{params.value?.split(".")[0]}</Link>;
+    header: "name",
+    accessorKey: "name",
+    size: 100,
+    Cell: ( { row } ) => {
+      const name = row.original.name;
+      return <Link href={`/nodes/${name}/`} color="inherit">{name?.split(".")[0]}</Link>;
     },
   },
   {
-    field: "machine_type",
-    width: 90,
+    header: "machine_type",
+    accessorKey: "machine_type",
+    size: 90,
+    maxSize: 90,
   },
   {
-    field: "up",
-    type: "boolean",
-    width: 70,
+    header: "up",
+    accessorFn: (row: Node) => row.up?.toLocaleString(),
+    size: 70,
+    filterVariant: "select",
   },
   {
-    field: "locked",
-    type: "boolean",
-    width: 70,
+    header: "locked",
+    accessorFn: (row: Node) => row.locked?.toLocaleString(),
+    size: 70,
+    filterVariant: "select",
   },
   {
-    headerName: "locked since",
-    field: "locked_since",
-    type: "date",
-    valueFormatter: (row: GridValueFormatterParams) => formatDate(row.value),
-    width: 150,
+    header: "locked since",
+    filterVariant: 'date',
+    sortingFn: "datetime",
+    accessorFn: (row: Node) => formatDate(row.locked_since),
+    size: 150,
+    enableColumnFilter: false,
   },
   {
-    headerName: "locked by",
-    field: "locked_by",
-    width: 175,
+    header: "locked by",
+    accessorKey: "locked_by",
+    size: 175,
+    filterVariant: "select",
   },
   {
-    headerName: "OS type",
-    field: "os_type",
-    width: 90,
+    header: "OS type",
+    accessorFn: (row) => row.os_type || "none",
+    size: 90,
+    filterVariant: "select",
   },
   {
-    headerName: "OS version",
-    field: "os_version",
-    width: 90,
+    header: "OS version",
+    accessorFn: (row) => row.os_version || "none",
+    size: 90,
+    filterVariant: "select",
   },
   {
-    field: "arch",
-    width: 60,
+    header: "arch",
+    accessorKey: "arch",
+    size: 60,
+    filterVariant: "select",
   },
   {
-    field: "description",
-    width: 500,
+    header: "description",
+    accessorKey: "description",
+    size: 500,
```

Comment:
Good feedback - I've pushed a commit that tightens up the node lists considerable. Let me know what you think!

Reply:
"description" text still seems to be cutting off. Maybe we can add a horizontal scroll or text-wrap description column because we can't say exactly how long the text could go. I like the text-wrap option because we do that in current pulpito too!

<img width="1725" alt="image" src="https://github.com/ceph/pulpito-ng/assets/50358181/ca383a78-76df-4617-bf9a-1087e82f7ca8">

Reply:
I'll see if I can get the text to wrap!